sampling/batch_creator.py

```diff
@@ -23,21 +23,28 @@ def run(self):
         """
         Keep creating the batches in the background
         """
+        cur_sample = None
         while not self.stop_batch_creator.is_set():
-            if self.ds.samples.empty():
-                print("Waiting for sample creator to create a sample")
-                continue
-            elif self.batches.full():
+            if self.batches.full():
                 print("Waiting for client to take a batch from batches array")
                 continue
+            elif cur_sample is None and self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
             else:
                 i = 0
-                if not curr_sample or self.offset == self.ds.sample_size:
-                    self.offset = 0
-                    curr_sample = self.ds.samples.get()
-                curr_batch = []
+                if cur_sample is None:
+                    cur_sample = self.ds.samples.get()
```

Comment:
Whenever you are picking a new sample set offset to 0

Reply:
I'm setting it in line 47.

Reply:
is there any improvement with this change or was there any error?

Reply:
Is there any improvement with this change or was there any error?

Reply:
There was an error. I will explain in person.

Reply:
okay